common/setups/rasr/gmm_system.py

```diff
@@ -1087,19 +1087,22 @@ def get_gmm_output(
                 corpus_key
             ].get(scorer_key, [None])[-1]
 
-        if corpus_type == "dev" and self.alignments[corpus_key].get(
-            f"alignment_{steps.get_step_names_as_list()[step_idx - 1]}", False
-        ):
+        if (corpus_type == "dev" or corpus_type == "test") and self.alignments[
+            corpus_key
+        ].get(f"alignment_{steps.get_step_names_as_list()[step_idx - 1]}", False):
             gmm_output.alignments = self.alignments[corpus_key][
                 f"alignment_{steps.get_step_names_as_list()[step_idx - 1]}"
             ][-1]
 
         if corpus_type == "train":
+            gmm_step_list = steps.get_gmm_steps_names_as_list()
+            step_list = steps.get_step_names_as_list()
+            dist_last_step = 1 + (len(step_list) - len(gmm_step_list))
```

Comment:
are you sure this logic works in any case, even if there are non_gmm_steps interleaved with gmm_steps?

e.g. `steps.get_step_names_as_list() == ["extract", "mono", "force_align", "tri", "force_align"]`

Reply:
You are totally right, this would not have worked. I changed the logic a bit and introduced `get_prev_gmm_step` which (at least with the test you proposed) worked. 
In general this makes `get_gmm_steps_names_as_list` kind of redundant and unused, but personally i would not mind to keep this function for potential future usage (since its already written) but this is by far no strong opinion so if you think this clutters the code I would just remove it.